### PR TITLE
Adding APIs and persistence for library status

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Store/SFSDKEventStoreManager.m [hunk 1]
```diff
@@ -103,6 +103,15 @@ - (void) storeEvents:(NSArray<SFSDKInstrumentationEvent *> *) events {
     }
 }
 
+- (NSInteger) numStoredEvents {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Store/SFSDKEventStoreManager.m]: New API for number of stored events.

File: libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Store/SFSDKEventStoreManager.m [hunk 2]
```diff
@@ -180,7 +189,10 @@ - (SFSDKInstrumentationEvent *) fetchEventFromFile:(NSString *) file {
     }
     NSData *data = self.dataDecryptorBlock([NSData dataWithContentsOfFile:file]);
     SFSDKInstrumentationEvent *event = [[SFSDKInstrumentationEvent alloc] initWithJson:data];
-    return [event copy];
+    if (event && event.eventId) {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Store/SFSDKEventStoreManager.m]: Tiny bugfix.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.h [hunk 3]
```diff
@@ -37,6 +37,7 @@
 @property (nonatomic, readonly, strong, nonnull) SFSDKEventStoreManager *eventStoreManager;
 @property (nonatomic, readonly, strong, nonnull) SFSDKAnalyticsManager *analyticsManager;
 @property (nonatomic, readonly, strong, nonnull) SFUserAccount *userAccount;
+@property (nonatomic, readonly, assign) BOOL enabled;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.h]: Exposing status of library.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m [hunk 4]
```diff
@@ -229,10 +236,29 @@ - (SFSDKDeviceAppAttributes *) buildDeviceAppAttributes {
     return [[SFSDKDeviceAppAttributes alloc] initWithAppVersion:appVersion appName:appName osVersion:osVersion osName:osName nativeAppType:appTypeStr mobileSdkVersion:mobileSdkVersion deviceModel:deviceModel deviceId:deviceId clientId:clientId];
 }
 
+- (void) storeAnalyticsPolicy:(BOOL) enabled {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: Adding persistence of `on/off`, just like `Android`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m [hunk 5]
```diff
@@ -229,10 +236,29 @@ - (SFSDKDeviceAppAttributes *) buildDeviceAppAttributes {
...
+        [self storeAnalyticsPolicy:YES];
+    }
+    self.enabled = [defs boolForKey:kAnalyticsOnOffKey];
+}
+
 #pragma mark - SFAuthenticationManagerDelegate
 
 - (void) authManager:(SFAuthenticationManager *) manager willLogoutUser:(SFUserAccount *) user {
     [self.analyticsManager reset];
+    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: Clearing stored status when logout occurs.

**Comment by wmathurin** [comment]
LGTM

**Comment by khawkins** [comment]
Next time, can we rebase before submitting for review?  It makes it a lot easier to go back through a PR later, in reference to a new point in time in the repo.

**Comment by bhariharan** [comment]
Yup, will do!

**Pull request merged**